src/main/java/io/Parser.java

```diff
@@ -0,0 +1,51 @@
+package io;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.NoSuchElementException;
+/**
+ * Used parse the additional data
+ * given in a command based on tags.
+ * @author Alvis Ng (supermii2)
+ */
+public class Parser {
+    /** Untagged input*/
+    private String defaultInput;
+    /** Dictionary of tagged inputs*/
+    private Dictionary<String, String> taggedInputs;
+    /**
+     * Creates a parsed representation
+     * of the input string given by
+     * the user's input.
+     * @param input
+     */
+    public Parser(String input) {
+        taggedInputs = new Hashtable<>();
+        String[] phrases = input.split("/");
+        this.defaultInput = phrases[0];
+        for (int i = 1; i < phrases.length; i++) {
+            String[] words = phrases[i].split(" ", 2);
+            taggedInputs.put(words[0], words[1]);
+        }
+    }
+    /**
+     * Get the untagged input
+     * @return Untagged input of parsed message
+     */
+    public String getDefaultString() {
```

Comment:
I think the naming could be more intuitive. It is not so clear what defaultString or defaultInput means. The JavaDoc could also be more descriptive.

Reply:
Thank you! I was struggling to come up with a name so came up with that. If you have any suggestions I would appreciate it! Maybe untagged or raw input.